src/common/utils/get_blueprint.py

```diff
@@ -18,6 +19,12 @@ def get_blueprint(self, type: str) -> Blueprint:
         resolved_reference: ResolvedReference = resolve_reference(
             type, lambda data_source_name: get_data_source(data_source_name, self.user)
         )
+        if is_reference(resolved_reference.entity):
+            resolved_reference = resolve_reference(
+                f"{resolved_reference.data_source_id}/{resolved_reference.entity['address']}",
```

Comment:
sometimes the address already contains a datasource. ex L70 in test_reference_resolve